app/controllers/comments_controller.rb

```diff
@@ -11,14 +11,16 @@ def edit
   def create
     @comment = Comment.new(comment_params)
     @comment.user = current_user
-
     respond_to do |format|
       if @comment.save
+        reference = @comment.root.commentable
         format.html { redirect_to :back, notice: 'Comment was successfully created.' }
         format.json { render :show, status: :created, location: @comment }
+        format.js { render( @comment.parent.nil? ? 'comment_on_reference.js.erb' : 'reply_to_comment.js.erb', locals: {reference: reference} )}
```

Comment:
so this is the part I was struggling with yesterday, but I've got a solution and wanted your opinion on it.  Basically because commenting on a reference is different enough than replying to a comment, I seperated out the logic into 2 different jquery pieces.  There's some duplication, so I was considering joining these two things with an if statement in the jquery, but this seems cleaner and I feel like it makes more sense to make the controller do that.  Thoughts?